CMakeLists.txt

```
@@ -88,37 +72,32 @@ ENDIF ()
 IF (MSGPACK_32BIT)
    IF ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
       SET (CMAKE_CXX_FLAGS "-m32 ${CMAKE_CXX_FLAGS}")
-      SET (CMAKE_C_FLAGS "-m32 ${CMAKE_C_FLAGS}")
       SET (CMAKE_EXE_LINKER_FLAGS "-m32 ${CMAKE_EXE_LINKER_FLAGS}")
-   ELSEIF ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+   ELSEIF ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
       SET (CMAKE_CXX_FLAGS "-m32 ${CMAKE_CXX_FLAGS}")
-      SET (CMAKE_C_FLAGS "-m32 ${CMAKE_C_FLAGS}")
       SET (CMAKE_EXE_LINKER_FLAGS "-m32 ${CMAKE_EXE_LINKER_FLAGS}")
    ENDIF ()
 ENDIF ()
 
 OPTION (MSGPACK_BUILD_EXAMPLES "Build msgpack examples." ON)
 
-IF (MSGPACK_BOOST)
-   SET (CMAKE_CXX_FLAGS "-DMSGPACK_USE_BOOST ${CMAKE_CXX_FLAGS}")
-   SET (Boost_USE_MULTITHREADED      ON)
-   SET (Boost_USE_STATIC_RUNTIME    OFF)
-   FIND_PACKAGE (Boost REQUIRED COMPONENTS chrono context system timer)
+SET (Boost_USE_MULTITHREADED      ON)
+SET (Boost_USE_STATIC_RUNTIME    OFF)
+FIND_PACKAGE (Boost REQUIRED COMPONENTS chrono context system timer)
```

```suggestion
FIND_PACKAGE (Boost REQUIRED)
```

When I removed components, then cmake find boost correctly on appveyor old msvc.
I think that omitting components is accepted solution.

---

Looks like `COMPONENTS` is necessary on Ubuntu and Mac.